dspace-api/src/main/java/org/dspace/external/provider/impl/SHERPAv2JournalDataProvider.java

```diff
@@ -0,0 +1,358 @@
+                        content.close();
+                    }
+                }
+            } else {
+                sherpaResponse = new SHERPAResponse("SHERPA/RoMEO returned no response");
+            }
+        } catch (Exception e) {
+            log.error("SHERPA/RoMEO query failed: ", e);
+        }
+
+        if (sherpaResponse == null) {
+            sherpaResponse = new SHERPAResponse(
+                "Error processing the SHERPA/RoMEO answer");
+        }
+        if (CollectionUtils.isNotEmpty(sherpaResponse.getJournals())) {
+            SHERPAJournal sherpaJournal = sherpaResponse.getJournals().get(0);
+
+            ExternalDataObject externalDataObject = constructExternalDataObjectFromSherpaJournal(sherpaJournal);
+            return Optional.of(externalDataObject);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Construct ExternalDataObject populated with journal metadata from the SHERPA v2 API response
+     * @param sherpaJournal
+     * @return external data object representing a journal
+     */
+    private ExternalDataObject constructExternalDataObjectFromSherpaJournal(SHERPAJournal sherpaJournal) {
+        // Set up external object
+        ExternalDataObject externalDataObject = new ExternalDataObject();
+        externalDataObject.setSource(sourceIdentifier);
+        // Set journal title in external object
+        if (CollectionUtils.isNotEmpty(sherpaJournal.getTitles())) {
+            String journalTitle = sherpaJournal.getTitles().get(0);
+            externalDataObject.setId(sherpaJournal.getTitles().get(0));
+            externalDataObject.addMetadata(new MetadataValueDTO(
+                "dc", "title", null, null, journalTitle));
+            externalDataObject.setValue(journalTitle);
+            externalDataObject.setDisplayValue(journalTitle);
+        }
+        // Set ISSNs in external object
+        if (CollectionUtils.isNotEmpty(sherpaJournal.getIssns())) {
+            String issn = sherpaJournal.getIssns().get(0);
+            externalDataObject.addMetadata(new MetadataValueDTO(
+                    "dc", "identifier", "issn", null, issn));
+
+        }
+
+        return externalDataObject;
+    }
+
+    /**
+     * Search SHERPA v2 API for journal results based on a 'contains word' query
+     * @param query The query for the search
+     * @param start The start of the search
+     * @param limit The max amount of records to be returned by the search
+     * @return a list of external data objects
+     */
+    @Override
+    public List<ExternalDataObject> searchExternalDataObjects(String query, int start, int limit) {
+        // query args to add to SHERPA/RoMEO request URL
+        HttpGet get = null;
+        try {
+            get = constructHttpGet(query, start, limit);
+            HttpClient hc = new DefaultHttpClient();
+            HttpResponse response = hc.execute(get);
+            if (response.getStatusLine().getStatusCode() == 200) {
```

Comment:
Can we use "HttpStatus.SC_OK" instead of 200 ?

Reply:
Done! See commit# 839159b